src/app/core/auth/state/auth-state.service.ts

```diff
@@ -1,24 +1,39 @@
 import { Action, State, StateContext } from '@ngxs/store'
 import { Injectable } from '@angular/core'
 import { AuthService } from '@core/auth/services/auth.service'
-import { Observable, tap } from 'rxjs'
+import { catchError, Observable, tap } from 'rxjs'
 import { Tokens } from '@core/auth/types/tokens'
 import { TokensStorageService } from '../services/tokens-storage.service'
-import { Login, LogOut, SetIsLoading, SetTokens } from './auth.actions'
+import {
+  Login,
+  LogOut,
+  Register,
+  SetIsLoading,
+  SetTokens,
+  ValidateEmail
+} from './auth.actions'
 import { Router } from '@angular/router'
+import { RegisterData } from '@core/auth/types/credentials'
+import { HttpErrorResponse } from '@angular/common/http'
 
 export interface AuthState {
   tokens?: Tokens
   isAuthenticated?: boolean
   loading?: boolean
+  username?: string
+  email?: string
+  validationEmailError?: string
 }
 
 @State<AuthState>({
   name: 'auth',
   defaults: {
     tokens: null,
     isAuthenticated: false,
-    loading: false
+    loading: false,
+    username: null,
+    email: null,
+    validationEmailError: null
   }
```

Comment:
No need to reset every field. You can just provide an empty object as default state and everything will be `undefined`.